src/permutation.rs

```diff
@@ -118,12 +117,14 @@ impl Permutation {
         // Pad remaining width with zero
         self.pad();
 
+        let fulll_rounds_iter = self.full_rounds / 2;
```

Comment:
This extra line does not seem to add anything. Having the two for loops go from `0...self.total_full_rounds / 2` seems readable

Reply:
It was not about readable, it was about just doing the division one time and make a call to a stack value twice. 

I know it won't make the difference, but it's more optimal and still being readable.
I can change it if you feel is better to.

Reply:
Probably a personal taste, for me it breaks the reading flow, and since the code calls the variable `iter` a reader may believe that it is an iterator. WDYT?

Reply:
Perfect. Let's keep it as you mention.